src/file_operations.rs

```diff
@@ -0,0 +1,97 @@
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+
+enum Op {
+    MkDir(PathBuf),
+    GitInit(PathBuf),
+}
+
+struct FileOperations {
+    root: PathBuf,
+    operations: Vec<Op>,
+}
+
+impl FileOperations {
+    fn rooted_at<P: AsRef<Path>>(path: P) -> FileOperations {
```

Comment:
Because `rooted_at` is a better name.  `new` is a good convention in general, but if there's a better name, I think it should be used instead.